extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/response/RestResponseResource.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.resteasy.reactive.server.test.response;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import org.jboss.resteasy.reactive.RestResponse;
+
+import io.smallrye.mutiny.Uni;
+
+@Path("")
+public class RestResponseResource {
+    @GET
+    @Path("rest-response")
+    public RestResponse<String> getString() {
+        return RestResponse.ok("Hello");
+    }
+
+    @GET
+    @Path("response-uni")
+    public Response getResponseUniString() {
+        return Response.ok(Uni.createFrom().item("Hello")).build();
+    }
+
+    @GET
+    @Path("rest-response-uni")
+    public RestResponse<Uni<String>> getUniString() {
+        return RestResponse.ok(Uni.createFrom().item("Hello"));
+    }
+
+    @GET
+    @Path("rest-response-json")
+    public JsonSomething getJson() {
+        return new JsonSomething("Stef", "Epardaud");
+    }
+
+    @GET
+    @Path("json")
+    public RestResponse<JsonSomething> getRestResponseJson() {
+        return RestResponse.ok(new JsonSomething("Stef", "Epardaud"));
```

Comment:
How about we also set some headers here so more parts of the `RestResponse` codepath can be tested

Reply:
Sure.

Reply:
Done.